src/atom.rs

```diff
@@ -539,22 +533,139 @@ where
 
 ///////////////////////////////////////////////////////////////////////////////
 
-fn encode_date_time(dt: &DateTime<Utc>) -> BytesText<'static> {
-    let s = dt.to_rfc3339_opts(chrono::SecondsFormat::Millis, true);
+fn encode_timestamp(
+    col: &Arc<dyn Array>,
+    index: usize,
+    unit: TimeUnit,
+    tz: Option<Arc<str>>,
+) -> Result<BytesText<'static>, UnsupportedDataType> {
+    let dt = match unit {
+        TimeUnit::Microsecond => {
+            let value = cast_primitive::<TimestampMicrosecondType>(col, index)?;
+            DateTime::from_timestamp_micros(value)
+        }
+        TimeUnit::Millisecond => {
+            let value = cast_primitive::<TimestampMillisecondType>(col, index)?;
+            DateTime::from_timestamp_millis(value)
+        }
+        TimeUnit::Nanosecond => {
+            let value = cast_primitive::<TimestampNanosecondType>(col, index)?;
+            Some(DateTime::from_timestamp_nanos(value))
+        }
+        TimeUnit::Second => {
+            let value = cast_primitive::<TimestampSecondType>(col, index)?;
+            DateTime::from_timestamp(value, 0)
+        }
+    };
+
+    match dt {
+        Some(d) => Ok(encode_date_time(&d, tz.is_some())),
+        None => Err(UnsupportedDataType::new(DataType::Timestamp(unit, tz))),
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+fn encode_date_time(dt: &DateTime<Utc>, tz: bool) -> BytesText<'static> {
+    let s = dt.to_rfc3339_opts(chrono::SecondsFormat::Millis, tz);
```

Comment:
I thought `tz: bool` here should indicate whether to add timezone to result or not, but you are passing it directly to [`to_rfc3339_opts`](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.to_rfc3339_opts) where second argument only controls whether it's OK to replace `+00:00` with `Z`, but timezone is still always present.

Reply:
Maybe if `tz == false` you should call [`naive_utc`](https://docs.rs/chrono/latest/chrono/struct.DateTime.html#method.naive_utc) to strip the timezone before formatting to string.